my-app/src/main/java/com/JavaWebServer/app/ClientWorkerService.java

```diff
@@ -1,33 +1,28 @@
 package com.JavaWebServer.app;
-import java.io.*;
-import java.net.*;
-import java.util.*;
 
 public class ClientWorkerService implements Runnable {
-  private Socket clientSocket = null;
+  private IClient client = null;
   private String serverName = null;
-  
-  public ClientWorkerService (Socket clientSocket, String serverName) {
-    this.clientSocket = clientSocket;
+
+  public ClientWorkerService (IClient client, String serverName) {
```

Comment:
There's no real convention that states you should prefix your interfaces with an `I`. I believe .NET does this, but in Java it's more common to see `interface Client` `class ClientImpl` which is not good either, as it adds nothing.

In this case, your `IClient` is the interface for `ClientSocket`, which I think is okay. However, I would recommend renaming `IClient` to `Client` or something even more descriptive, as `Client` doesn't really explain that it is an interface for a `Socket`'s implementation